test/core/OriginCore.js

```diff
@@ -0,0 +1,263 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Test: OriginCore.js
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const eventsDecoder = require('../lib/event_decoder.js');
+
+const OriginCore = artifacts.require('OriginCore');
+const MockToken = artifacts.require('MockToken');
+
+contract('OriginCore', async (accounts) => {
+    describe('reporting a block', async () => {
+        let ost;
+        let originCore;
+
+        beforeEach(async () => {
+            ost = await MockToken.new();
+            originCore = await OriginCore.new(3, ost.address);
+        });
+
+        it('should accept a correct block report', async () => {
```

Comment:
please also assert that the balance of the core contract is increased with `COST_REPORT_BLOCK`